### ISSUE TITLE
Unable to see at a glance what filters I am using

### ISSUE THREAD

**Issue by BendixSpring** [TeamMentor/TM_4_0_Design#348]
If I click on the Android filter, I get 24 articles.  The Guidance term is highlighted, but I have to scroll to the very bottom to see that the Android filter is bold.  I think you should put the active filters on the Navigate bar so folks have some context.
![image](https://cloud.githubusercontent.com/assets/3578332/5946647/62159b00-a705-11e4-8477-fcb6d2e595cb.png)

**Comment by DinisCruz** [comment]
@luchia can you take this on?

**Comment by luchia** [comment]
Yes :)

**Comment by luchia** [comment]
@roman87 @BendixSpring @DinisCruz @michaelhidalgo @tekgirl77, tell me what you think about this:

![screen shot 2015-01-29 at 11 24 02 am](https://cloud.githubusercontent.com/assets/4547737/5951114/18297b20-a7ae-11e4-9208-977c76b490a5.png)

Related to #299.

**Comment by luchia** [comment]
This gif also relates to this issue and a number of conversations we've had during our daily meetings about the Filters div being exessively long. I built a custom dropdown system with CSS only, let me know what you think:

![filters](https://cloud.githubusercontent.com/assets/4547737/5951148/a5b1608e-a7ae-11e4-8fa9-ff4458ed1abc.gif)

@DinisCruz - I mostly got this working except the links don't actually work for some reason.

**Comment by BendixSpring** [comment]
That seems fine.  What is the difference between a filter and the box above showing Type / Vulnerability?

How did you get Vulnerability to appear on the left under the logo?  I only was able to get Guidance displayed.

**Comment by BendixSpring** [comment]
This looks good.  Can you have more than one filter selected?  If so, will you be able to see them all?

**Comment by tekgirl77** [comment]
I think that looks great Lucy!  I like it.


On Wednesday, January 28, 2015, Luchia Bloomfield <notifications@github.com>
wrote:

> This gif also relates to this issue and a number of conversations we've
> had during our daily meetings about the Filters div being exessively long.
> I built a custom dropdown system with CSS only, let me know what you think:
>
> [image: filters]
> <https://cloud.githubusercontent.com/assets/4547737/5951148/a5b1608e-a7ae-11e4-8fa9-ff4458ed1abc.gif>
>
> @DinisCruz <https://github.com/DinisCruz> - I mostly got this working
> except the links don't actually work for some reason.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TeamMentor/TM_4_0_Design/issues/348#issuecomment-71955950>
> .
>


-- 

Salle Ingle
Support & Implementation Specialist, eLearning
[image: https://www.securityinnovation.com/_img/art/logo.png]

**Comment by luchia** [comment]
Further work on the style with this issue. Also @DinisCruz, I got the links to work in the dropdowns. I couldn't work out how to get the active filter into the 'Filtered by:' badge though.

![screen shot 2015-01-29 at 12 49 49 pm](https://cloud.githubusercontent.com/assets/4547737/5951613/58a2ff30-a7b5-11e4-89ef-9448b22f9682.png)
![screen shot 2015-01-29 at 12 49 51 pm](https://cloud.githubusercontent.com/assets/4547737/5951615/58a5da7a-a7b5-11e4-9fee-995a19318b12.png)
![screen shot 2015-01-29 at 12 49 53 pm](https://cloud.githubusercontent.com/assets/4547737/5951614/58a4adda-a7b5-11e4-8645-b50eaf413af1.png)

@BendixSpring so the box showing the Type / Vulnerability path is the sub-navigation that acts in a similar way to breadcrumbs. It allows users to go back to a previous directory.

I got to those pages by starting [here](http://localhost:1337/-/), then I clicked Type, then Vulnerabilities. You may or may not have this routing on the beta yet.

On Jade you cannot have more than one filter selected but on Flare, you can :)

**Comment by DinisCruz** [comment]
Interesting CSS trick, does it work on all browsers?

**Comment by luchia** [comment]
Should do, it only uses the :hover pseudo-element. I just checked [caniuse](http://caniuse.com/#search=%3Ahover) and it says supported by legacy browsers.

**Comment by DinisCruz** [comment]
@michaelhidalgo is this something we can test on sause labs?

**Comment by roman87** [comment]
Lets not mix two issues together. Remember this is still an issue on the clue to the user on what filter is selected.  @BendixSpring  on this version you will be able to select only one filter. The word "Vulnerabilities" comes from a different library that Lucy is using in QA.

**Comment by roman87** [comment]
@DinisCruz can you wire up the "active filters" button? .

**Comment by DinisCruz** [comment]
Here is the test that confirms that the active filter shows **Filter here**
```coffee
  it 'Check Active Filter',(done)->
    jade.page_User_Graph 'Java', (html,$)->
      activeFilter = $('#activeFilter').text()
      activeFilter.assert_Is 'Filter here'
      done()
```

**Comment by DinisCruz** [comment]
Active filter is now wired, and here is the test that checks for it:

```coffee
  it 'Check Active Filter',(done)->
    jade.page_User_Graph 'Java', (html,$)->
      activeFilter = $('#activeFilter').text()              # on first load there should be no value in the active filter
      activeFilter.assert_Is ''
      first_Filter = $('#filters a').first()
      link = first_Filter.attr().href
      page.open link, (html, $)->
        updatedFilter = $('#activeFilter').html()
        first_Filter.text().assert_Contains updatedFilter   # now active filter should have the contents of the link clicked
        done()
```

Here is what it looks where there is no filter selected:

![image](https://cloud.githubusercontent.com/assets/656739/6007344/3849cc82-ab11-11e4-8ade-67eb03cfbec7.png)

Here it is with the active filter:

![image](https://cloud.githubusercontent.com/assets/656739/6007360/4fd270ac-ab11-11e4-8d6f-0d6eabd728dd.png)

@luchia I made the selected filter bold, so that there was a good visual clue to show which one was selected, but it should be easy for you to change it (if you want). 

We also need to think about how to clear the filters. 

At the moment this happens if any of the navigation links is clicked (or left-hand-side links) or if the selected filter is clicked (the one in bold)

**Comment by roman87** [comment]
Which branch is it on? I don't see the changes?

**Comment by roman87** [comment]
Related to #373 #374 #375

**Comment by DinisCruz** [comment]
Just pushed fix into Dev of (TM_Design) and master (of TM_QA)

**Comment by roman87** [comment]
Clicking on a filter now crashes TMDesign :)

**Comment by BendixSpring** [comment]
This is great!  One other suggestion:  That filter box is taking up lots of room and has more text than someone might want.  Can you put some sort of toggle to make it hide/unhide?

**Comment by roman87** [comment]
@BendixSpring that is exactly the idea in Flare :) both the lefthand and the righhand side filters will slide in and out.

**Comment by BendixSpring** [comment]
Although I generally run with JavaScript turned off on most sites, I do turn it on for some trusted sites where functionality breaks.

 

What is the business rationale for making a JS version and non-JS version?  My guess is that most modern sites with any level of functionality require JS.

**Comment by DinisCruz** [comment]
this is now an @luchia issue (as far as I can see)

**Comment by luchia** [comment]
As per yesterdays team meeting, there will be two layouts for filters that need to be A/B tested. The first is as per this entire thread (and included below for ease of reference):

![screen shot 2015-02-05 at 9 40 01 am](https://cloud.githubusercontent.com/assets/4547737/6052141/fc2e5126-ad1a-11e4-83b4-3c979295ec90.png)
![screen shot 2015-02-05 at 9 38 48 am](https://cloud.githubusercontent.com/assets/4547737/6052120/d32a5a4a-ad1a-11e4-9d93-a09cba3c250b.png)
![screen shot 2015-02-05 at 9 38 51 am](https://cloud.githubusercontent.com/assets/4547737/6052123/d32c3fea-ad1a-11e4-8c9e-305d967ea9a4.png)
![screen shot 2015-02-05 at 9 38 44 am](https://cloud.githubusercontent.com/assets/4547737/6052122/d32af3ec-ad1a-11e4-9744-7e4f52030b5f.png)

There is a known bug with this resolution though. When the user moves the mouse downwards they can skip the next button because of the way the CSS works. You can see how that works below:

![filters](https://cloud.githubusercontent.com/assets/4547737/6052180/7e4e5aa2-ad1b-11e4-9403-fce4ef6a66ef.gif)

We made a team decision to come up with an alternative solution that didn't rely on :pseudo classes to create dropdown buttons. This solution can be seen below and can be found in the POC directory.

![screen shot 2015-02-05 at 10 55 57 am](https://cloud.githubusercontent.com/assets/4547737/6053002/9fabf286-ad25-11e4-9bb5-e59031a7b945.png)

**Comment by roman87** [comment]
This is fixed. Will follow up with @BendixSpring when next beta is out.

**Comment by roman87** [comment]
@BendixSpring this has been fixed on latest beta at https://beta.teammentor.net could you take a look and let us know if the fix is as expected. Note that currently you can only select one filter at a time.

**Comment by BendixSpring** [comment]
When you first login, you are given essentially a blank screen.  Ok. Not blank, but pretty bare compared to the rest of the screens.  So in order to even see any filters, you have to click NAVIGATE.  This seems to be an unnecessary step.

 

So when I select Java, I see that Java is highlighted in the Filter By panel and is listed in the Showing xxx articles panel.  Nice!  But where is the search box?  I have to click the search box?  This seems like another unnecessary step.

 

Is still think the Filter by panel has too much stuff.  Scrolling is awkward.

 

Also, in order to set a filter, you have to click on the actual text.  It would be simpler if the user could click anywhere in the filter box to set a filter.

 

Finally, unless you can select multiple filters, don’t bother.  The filtering and searching must be at least as sophisticated as google.

 

--Danny

**Comment by roman87** [comment]
@BendixSpring Thank you for this excellent feedback. I've updated your comment with specific issues where each of these is tracked. Regarding this particular issue, given your comment " So when I select Java, I see that Java is highlighted in the Filter By panel and is listed in the Showing xxx articles panel. Nice!" I assume it has been solved to your satisfaction. I am closing this issue then. Let me know if this is not the case.
